uiTests/src/test/resources/scenarios-and-utils.rhino.js

```diff
@@ -11,113 +12,135 @@ importClass(com.intellij.openapi.actionSystem.AnActionEvent);
 importClass(com.intellij.openapi.actionSystem.Presentation);
 importClass(com.intellij.openapi.application.ApplicationManager);
 importClass(com.intellij.openapi.application.ModalityState);
+importClass(com.intellij.openapi.progress.ProgressManager);
 importClass(com.intellij.openapi.wm.ToolWindowId);
 importClass(com.intellij.openapi.wm.ToolWindowManager);
 importClass(com.intellij.ui.GuiUtils);
 
+// Note that due to how Remote Robot operates, each `runJs`/`callJs` invocation is executed in a fresh JavaScript environment.
+// Thus, we can't store any state between invocations.
 
 // Do not run in EDT.
 function sleep() {
   Thread.sleep(500);
 }
 
-// Do not run in EDT.
-function configureIde() {
-  const settings = GeneralSettings.getInstance();
-  settings.setConfirmExit(false);
-  settings.setShowTipsOnStartup(false);
+// Do not run any of the methods in EDT.
+function Ide() {
+  this.configure = function () {
+    const settings = GeneralSettings.getInstance();
+    settings.setConfirmExit(false);
+    settings.setShowTipsOnStartup(false);
+  };
+
+  this.openProject = function (path) {
+    ProjectUtil.openOrImport(path, /* projectToClose */ null, /* forceOpenInNewFrame */ false);
+  };
+
+  this.soleOpenedProject = function () {
+    return new Project(ProjectUtil.getOpenProjects()[0]);
+  }
+
+  const getMethod = function (clazz, methodName) {
+    try {
+      return clazz.getDeclaredMethod(methodName);
+    } catch (e) {
+      if (e.javaException instanceof NoSuchMethodException) {
+        return getMethod(clazz.getSuperclass(), methodName);
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  this.awaitNoBackgroundTask = function () {
+    const progressManager = ProgressManager.getInstance();
+    const method = getMethod(progressManager.getClass(), 'getCurrentIndicators');
```

Comment:
Apparently no public API gives access to ongoing background tasks :frowning_face: